### ISSUE TITLE
Consider scrapping AnnotateDataset repartitioning

### ISSUE THREAD

**Issue by MattWellie** [populationgenomics/production-pipelines#981]
https://github.com/populationgenomics/production-pipelines/blob/8deea35191aff5c303f5bfab5f351ac2f11146c9/cpg_workflows/query_modules/seqr_loader.py#L229-L232

Low priority: Maybe we should consider removing this downsampling prior to AnnotateDataset

Original Issue: When AnnotateCohort was moved off DataProc, Hail's Query backend had issues ingesting VCFs above a certain size. We overcame this by setting a block size limit in `import_vcf` (the #MB per partition) which was pretty low. I don't remember all the reasoning here, just that I played with the number until the pipeline stopped breaking.

Next Issue: With a low block size and a high VCF size, we generated MASSIVELY fragmented datasets (115000+ partitions in our genome callsets). This did not cause problems in QoB, but became impossible to process in a local Hail Query environment (i.e. Talos). To improve usability, the block size override was removed (Hail Query has not broken), AND an extra patch was put into AnnotateDataset - when we break the huge callset into smaller per-project MTs, we do a proportional downsampling of the partitions.

In theory this is logical, with fewer partitions we gain downstream by giving Hail less busywork as there are fewer QoB jobs to keep track of. By repartitioning per-subset we're also making sure that each partition is data-dense, making good use of the spark parallelisation.

This downsampling does not break anything, but the process that extracts single-dataset MTs from the bigger MT takes hours. Contradicting Hail team comments/documentation, this is not a 'computationally free' process, it's expensive in terms of time and money 😬

This Issue: Now that we're not overriding the block sizes when we read in the 1TB+ VCF in AnnotateCohort (since https://github.com/populationgenomics/production-pipelines/pull/882), the whole-genome MT we're storing from AnnotateCohort is ~7700 partitions. That's still a lot, but way lower than the 115000 we were generating previously. Crucially, at 7700 partitions we have no issues using the data downstream (either in Talos, or when generating ES indexes), so there's no need to repartition for even more efficiency gains.

Example batch: https://batch.hail.populationgenomics.org.au/batches/529253

This is extracting 34 samples from a large callset, and has been running for 2+ hours. We haven't even created the MT to run AnnotateDataset on in that time :/ I haven't dug back into the archives to find out how long this ran for in DataProc, but that's probably not a fair comparison - DP jobs ran quickly, at great expense...

---

TL; DR - Now that AnnotateCohort creates MTs with a reasonable number of partitions, we can investigate whether the time/cost of downstream repartitioning is a waste. I don't think we have a clear test of AnnotateDataset with vs. without this repartitioning, but on the face of it this appears to be taking a massive amount of time.
